src/loading/shaderpack/shaderpack_validator.cpp

```diff
@@ -2,6 +2,8 @@
 
 #include <fmt/format.h>
 
+#include <WinNls.h>
```

Comment:
What is this header?

Reply:
It gives me the ability to use GetUserDefaultLangID()

Reply:
That's not helpful

Where is it from? What does it provide? What other solutions have you considered and why did you go with this one?

Reply:
I have not tried any other solutions. What it does is gives me the current user’s default language so that I can pull that from the language json. I plan to use a different solution in the future as this one is very far from cross-platform, but for now this is a temporary solution for testing, given I work in windows

Reply:
I believe the header itself lets me pull quite a few helpful options from a windows user’s settings

Reply:
Oh, so the proposal is more about the json file and how data is stored there? That would be a great thing to mention in the PR description

I'd recommend looking at c++ localization libraries. I don't know what's available and how well it'd fit Nova, but we won't know until we look.

Reply:
Also, how would this handle on other operating systems? Wouldn't you need to special case this based on current OS so that there's a way to get the user lang for Linux as well?

Reply:
Actually, it looks like this might help: https://stackoverflow.com/questions/3702665/how-can-i-get-the-system-language-in-c-c

I think `std::locale("");` should work; apparently using a nameless locale gives you the system default :)

Reply:
I wasn’t looking for a cross platform way of doing things just yet. Right now I’m using that header as a base testing method, and I’ll do research on cross platform alternatives when I get a chance

Reply:
Ah, fair enough. Well, for reference in the future:

**Linux:**
This returns the system locale, based on the `LANG` environment variable.

`setlocale(LC_ALL, NULL);`

**macOS:**
You need a specific header for macOS.

https://stackoverflow.com/questions/39255590/how-to-get-the-locale-of-the-current-user-in-osx-using-c